### ISSUE TITLE
Dynamic resource limits

### ISSUE THREAD

**Issue by fernandocamargoti** [spotify/luigi#2492]
Right now, it's possible to specify the resources needed by the Task dynamically. But it's not possible to set the limit dynamically, since it must be fixed in the config file.

In my case, the resources are cuda devices. I'll run the same code in machines with no cuda devices, single cuda devices or many cuda devices. So, I don't know in advance how many I have. Right now, I need to update the config file. I'd like to use torch.cuda.device_count() (an utilitaty method from PyTorch) to set the amount of cuda devices I have.

**Comment by dlstadther** [comment]
I'll be happy to review a PR! Feel free to submit! :)

**Comment by Tarrasch** [comment]
The only non-trivial part is that you're setting a per-scheduler configuration now instead of a perk-task configuration. Should still be doable, probably even easier.

**Comment by fernandocamargoti** [comment]
How could it be implemented? Right now, the config files are .cfg. Couldn't luigi support python files as well? I see lots of other projects using python files for configuration. That could add the dynamism we need.

Anyway, here is my current workaround:

I've created a wrapper around luigid as follows:

```
#!/usr/bin/env python

import os
import torch

if __name__ == "__main__":
    luigi_resources_path = os.path.join("output", "luigi_resources.cfg")

    with open(luigi_resources_path, "w") as f:
        f.write(f"[resources]\ncuda={torch.cuda.device_count()}")

    os.environ["LUIGI_CONFIG_PATH"] = luigi_resources_path

    from luigi.cmdline import luigid
    luigid()
```
